wolfcrypt/test/test.c

```diff
@@ -14407,34 +14481,44 @@ static int dh_test_ffdhe(WC_RNG *rng, const DhParams* params)
 
 #endif /* HAVE_FFDHE */
 
-int dh_test(void)
+static int dh_test(void)
 {
     int    ret;
     word32 bytes;
     word32 idx = 0, privSz, pubSz, privSz2, pubSz2;
-    byte   tmp[1024];
-#if !defined(USE_CERT_BUFFERS_3072) && !defined(USE_CERT_BUFFERS_4096)
-    byte   priv[256];
-    byte   pub[256];
-    byte   priv2[256];
-    byte   pub2[256];
-    byte   agree[256];
-    byte   agree2[256];
-#else
-    byte   priv[512];
-    byte   pub[512];
-    byte   priv2[512];
-    byte   pub2[512];
-    byte   agree[512];
-    byte   agree2[512];
-#endif
+    byte   *tmp = NULL,
```

Comment:
Would it be possible to gate using heap here vs stack variables based on `WOLFSSL_SMALL_STACK`? There might be cases where it would be nice for the test to use stack vs heap.

Reply:
very easy to do, but there's a slew of these stack->heap refactors.  I can go back over all of them and add stack allocations gated on `-UWOLFSSL_SMALL_STACK` if you think it's worth the trouble.  if I do, it will be clean, as I'll just add `_stackbuf` arrays in the decls section, and replace the `XMALLOC()`s with assignments to the `_stackbuf` variants.  thinking yes?

Reply:
update:  done.  found a neat way to do it.  and found and fixed a bug while I was at it:
```
static int dh_test(void)
{
     int    ret;
     word32 bytes;
     word32 idx = 0, privSz, pubSz, privSz2, pubSz2;
    byte   *tmp = NULL,
      *priv = NULL,
      *pub = NULL,
      *priv2 = NULL,
      *pub2 = NULL,
      *agree = NULL,
      *agree2 = NULL;
    word32 agreeSz = (word32)sizeof(agree);
    word32 agreeSz2 = (word32)sizeof(agree2);
```
d'oh!  lucky find.